### DISCUSSION TITLE
Python Wrapper Custom Displacements not Working Properly in  Parallel

### DISCUSSION THREAD

**Discussion by kursatyurt** [su2code/SU2#2052] [General]
I am trying to deform a NACA0012 Airfoil using the Python interface, I am skipping the halo nodes at the interface assuming `CommunicateMeshDisplacements`  handle communication as halo nodes can be problematic in my case. The code works well on a single core but creates problems at the boundaries between processors when run on multi-core as seen below. If I plot the undeformed coordinates and displacement individually, I don't have a problem. 

The total number of base points for both single-core and multi-core cases is 798 points. I have checked that incase I am skipping some base points.

![Deformed Mesh](https://github.com/su2code/SU2/assets/57598663/e6272471-caff-4554-b906-30175d8f3c35)

![Base points and displaced locations](https://github.com/su2code/SU2/assets/57598663/1da66d8c-3694-4a4f-9d28-9e5edfb25418)

I use following loop to get physical point:

```python
nVertex_MovingMarker = FluidSolver.GetNumberMarkerNodes(MovingMarkerID)
nVertex_MovingMarker_HALO = 0  # FluidSolver.GetNumberHaloNodes(MovingMarkerID)

        # Obtain indices of all vertices that are being worked on on this rank
for iVertex in range(nVertex_MovingMarker):
    iPoint = FluidSolver.GetMarkerNode(MovingMarkerID, iVertex)
    if FluidSolver.GetNodeDomain(iPoint):
        iVertices_MovingMarker_PHYS.append(int(iPoint))
    else:
        nVertex_MovingMarker_HALO += 1

nVertex_MovingMarker_PHYS = nVertex_MovingMarker - nVertex_MovingMarker_HALO
```

and later it deformed in the following part

```python

for i, iVertex in enumerate(iVertices_MovingMarker_PHYS):
FluidSolver.SetMarkerCustomDisplacement(MovingMarkerID, i, displacements[i, :])

FluidSolver.CommunicateMeshDisplacements()
FluidSolver.Preprocess(0)

FluidSolver.Run()

**Comment by pcarruscag** [comment]
You have a bug in your code, you are storing point indices but using them as vertices.
Always loop over all vertices and skip the ones for which the associated point is an halo.

**Reply by kursatyurt** [reply]
Thanks, I fixed it since I am taking two different codes as reference I mixed up the stuff.
